### ISSUE TITLE
Change in the API of Scanamo client API

### ISSUE THREAD

**Issue by sahlone** [scanamo/scanamo#174]
We can see `Scanamo` client uses this return type:
`scala.Option[scala.Either[com.gu.scanamo.error.DynamoReadError, T]]`
The better synatax will be 
`scala.Either[com.gu.scanamo.error.DynamoReadError, scala.Option[T]]`
It will help in better error handling and better syntax as `Either` is right biased now.
If we can take it a step further:
`scala.Either[com.gu.scanamo.error.ScanamoError, T]`
`trait ScanamoError(msg;String , ex:Exception)`

**Comment by philwills** [comment]
I agree, this is largely a historical hangover and I'd like to overhaul the API to much more of treating errors as values, but it would be a big change, likely effectively Scanamo 2.x and I'm not finding the time for such a big change at the moment.

**Comment by sahilahmadlone** [comment]
If you can come up with plan perhaps I can help.
